ext/ddtrace_profiling_native_extension/collectors_cpu_and_wall_time_worker.c

```diff
@@ -363,11 +363,17 @@ static void *run_sampling_trigger_loop(void *state_ptr) {
     state->stats.trigger_sample_attempts++;
 
     // TODO: This is still a placeholder for a more complex mechanism. In particular:
-    // * We want to signal a particular thread or threads, not the process in general
     // * We want to track if a signal landed on the thread holding the global VM lock and do something about it
     // * We want to do more than having a fixed sampling rate
 
-    kill(getpid(), SIGPROF);
+    current_gvl_owner owner = gvl_owner();
+    if (owner.valid) {
+      pthread_kill(owner.owner, SIGPROF);
+    } else {
+      // TODO: This is not a great "plan B", will be improved on a later PR
+      kill(getpid(), SIGPROF);
```

Comment:
Not a great plan B but identical to what was before, so I guess it's fine.

Reply:
Yeah -- I wanted to leave something in, but definitely this is the next PR I'm working on 😅